### DISCUSSION TITLE
bindings for Copying previous link or opening the previous links in new tab

### DISCUSSION THREAD

**Discussion by txtyash** [gdh1995/vimium-c#967] [Q&A] [answered]
Let's say I went from webpage A to webpage B in the same tab and started editing a form on webpage B. Now if I want to go back to the previous webpage A I'll lose the data on webpage B. This can be solved if there was a keybinding to either copy the link to the previous webpage or open the previous webpage in a new tab.

**Comment by gdh1995** [comment, marked as answer]
You may try this mapping:
```
# open as a new active tab
map <a-s-a> goBack reuse="newFg"
```

And `reuse` can also be `newBg`, which means to open the previous page in a new and background (inactive) tab.

**Reply by txtyash** [reply]
Works well!

**Comment by gdh1995** [comment]
BTW, there's no browser API to "get URL of the previous page" without opening it in a existing tab.

**Reply by txtyash** [reply]
would saving the history somewhere be a heavy task? This could also be turned into a vomnibar like menu..... like pressing a keybinding would open up vomnibar with entire history of the current tab.

**Reply by gdh1995** [reply]
yes too expensive and it may conflict with how your browser remembers or auto forgets those links.
